### PR TITLE
save git status & SHA1

### PR DESCRIPTION
**Pull request by davharris**
Given the fun we've been having with #62, I thought it would be a good idea to save the states of our repositories at the moment we save stuff to the `data` folder.

Currently, I'm saving the following information:

```
Local:    master /Users/davidharris/projects/BBSforecasting/
Remote:   master @ upstream (https://github.com/weecology/bbs-forecasting.git)
Head:     [06d7d4f] 2016-08-11: Merge pull request #63 from sdtaylor/fix_min_num_years
```

Plus a list of files that have changed since HEAD (divided into staged/unstaged/untracked).

If this looks good to you two, I can start adding similar code elsewhere in the repo.  If you have other suggestions, I'd be happy to incorporate those first.

### CODE REVIEW

**Comment by sdtaylor** [comment]
might also be good to save the md5sum of files with tools::md5sum. I'm not sure if that would work with sqlite files though. 
in consolidating  databases in calls in #38 I made a single function that should handle it all, so this can probably go in there just once.

**Comment by davharris** [comment]
I think @henrykironde had some issues with consistently hashing sqlite files across platforms. Maybe he can chime in?

**Comment by henrykironde** [comment]
will check this out  later

**Comment by sdtaylor** [comment]
if we reaaaaally want to keep track of the sqlite file I think we can do an sql dump and hash that.

**Comment by davharris** [comment]
I get `ad35459e5c013fea6f102bbad61ff269612a7bf6` as my SHA1 for `richness_w_env.txt`. @sdtaylor can you confirm?

If this looks good to you, I think it might be good enough for now.

**Comment by sdtaylor** [comment]
It does look to be working good but I'm hitting a snag where the servers for downloading elev data are having issues. Will try again later.

**Comment by sdtaylor** [comment]
k, it's working again. my SHA1 for `richness_w_env.txt` is `b772e4027da068779f8ed7753ee87340863f0c1c`. 

The others are: 
`env_data.txt` `8ba3ff664b24136ce8445c9f971b28c8d79cf9ec`
`gimss_ndvi_bbs.txt` `3455e553c0e73a681dc679ffd0d6468aa2c7ae74`

**Comment by davharris** [comment]
@sdtaylor is it possible that we're running into non-deterministic problems because of finicky downloads and different NA values across calls to the servers?

**Comment by ethanwhite** [comment]
> @sdtaylor is it possible that we're running into non-deterministic problems because of finicky downloads and different NA values across calls to the servers?

That would explain why I had different nulls in #62, which I never did understand.

**Comment by davharris** [comment]
Okay, I guess that's the next thing to test.  😲

**Comment by ethanwhite** [comment]
Data - it's never as easy as it should be.

**Comment by davharris** [comment]
@ethanwhite Want to try running this branch and seeing if the hashes match Shawn's and/or mine?

**Comment by sdtaylor** [comment]
I think it's that we're on different operating systems. There could be issues with floating points due to that. Checkout the the [`digest` vignette](https://cran.r-project.org/web/packages/digest/vignettes/sha1.html).

Maybe switch to `digest::sha1()` instead of `digest::digest()`. Those two give me different results because `sha1()` truncates the floating points.

```
> devtools::load_all()
Loading BBSforecasting
> x=get_richness_ts_env_data(1982, 2013,25)
> digest::digest(x, 'sha1')
[1] "b772e4027da068779f8ed7753ee87340863f0c1c"
> digest::sha1(x)
[1] "918ea37daadea77735aec649126cfc7fdbe3d5af"
```

**Comment by davharris** [comment]
Wait a minute.  The SHA-1 standard says everything should be done with integers, right? Every bit string has a single correct SHA-1 hash, right? 

What is going on?

**Comment by davharris** [comment]
Or, are you saying that the problem is that our computers could be representing the same floating point values with different bit strings? That would at least make sense, although it's still kind of horrifying.

**Comment by davharris** [comment]
Looks like that wasn't the problem :-(

```
> x=get_richness_ts_env_data(1982, 2013,25)
> digest::digest(x, 'sha1')
[1] "ad35459e5c013fea6f102bbad61ff269612a7bf6"
> digest::sha1(x)
[1] "8902be3896b549505938ba6723ac9258e5b9a318"
```

**Comment by davharris** [comment]
Thanks to some excellent sleuthing by @sdtaylor, I think we're all set here (apparently Retriever used an old version of the BBS data on my machine somehow).

Anything else to deal with before we merge?

**Comment by sdtaylor** [comment]
@davharris show us some hashes! :smiley:

**Comment by ethanwhite** [comment]
Happy to help here now if there are still things for me to check. Just give me details on what to run and I'll get it done.

**Comment by sdtaylor** [comment]
There are way too many merge conflicts on this now. Probably best to do it over.

**Comment by ethanwhite** [comment]
> There are way too many merge conflicts on this now. Probably best to do it over.

Ah, yeah, the whitespace changes are a killer.

**Comment by ethanwhite** [comment]
Actually, using  `git merge -Xignore-space-change` makes this really manageable. Just 3 conflicts total in two files. If/when we're happy with it I'm happy to merge it manually.

**Comment by davharris** [comment]
Sorry again about the whitespace.  I'd turned that setting off globally, but apparently it didn't propagate to projects that had already been created like this one.  It's really turned off now, so it shouldn't happen again.

**Comment by sdtaylor** [comment]
dave I put in a PR for you branch that fixes these conflicts

**Comment by davharris** [comment]
Ready to merge. Thanks @sdtaylor

File: R/forecast-bbs-core.R [hunk 1]
```diff
@@ -255,18 +261,20 @@ get_pop_ts_env_data <- function(start_yr, end_yr, min_num_yrs){
 get_richness_ts_env_data <- function(start_yr, end_yr, min_num_yrs){
   get_pop_ts_env_data(start_yr, end_yr, min_num_yrs) %>%
```
Comment (sdtaylor) [inline R/forecast-bbs-core.R]: sorry, one last think I missed. This should be   
```
 richness_ts_env_data = get_pop_ts_env_data(start_yr, end_yr, min_num_yrs) %>%
     collapse_to_richness()
  ```

Comment (davharris) [inline R/forecast-bbs-core.R]: Like this? https://github.com/weecology/bbs-forecasting/pull/64/commits/3da5e8905a1e4d7e6e94be05ae3847329a08d0d6

**Pull request merged**